packages/flutter_tools/test/integration.shard/test_utils.dart

```diff
@@ -19,11 +19,16 @@ Directory createResolvedTempDirectorySync(String prefix) {
   return globals.fs.directory(tempDirectory.resolveSymbolicLinksSync());
 }
 
-void writeFile(String path, String content) {
-  globals.fs.file(path)
+void writeFile(String path, String content, { bool sendToFuture = false }) {
+  final File file = globals.fs.file(path)
     ..createSync(recursive: true)
-    ..writeAsStringSync(content)
-    ..setLastModifiedSync(DateTime.now().add(const Duration(seconds: 10)));
+    ..writeAsStringSync(content);
+  // When making edits for hot reload changes, the synchronous file write combined
+  // with the immediate hot reload request may arrive too close together
+  // Ensure the change is recognized by sending it into the future.
```

Comment:
Huh. Wouldn't the edit need to go further into the past to ensure it is picked up as being before the hot reload request?

Reply:
The logic is if the modified timestamp occurred after the last compile, the file is considered dirty. So we can either send the edit into the future, or the last compile into the past

Reply:
Ah okay. That explanation would be good to have in the comment.